docs/messaging/messaging-metrics.md

```diff
@@ -264,6 +267,34 @@ _Note: The need to report `messaging.process.messages` depends on the messaging
 <!-- END AUTOGENERATED TEXT -->
 <!-- endsemconv -->
 
+### Metric: `metric.messaging.latency.duration`
+
+This metric is [recommended][MetricRecommended] for any messaging system with the capability to extract these timings. It represents the observed time between when a message was created, for example, by a producer, to the time it began being processed, for example, by a consumer.
+
+This metric SHOULD be specified with
+[`ExplicitBucketBoundaries`](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.31.0/specification/metrics/api.md#instrument-advice)
+of `[ 0.01, 0.05, 0.1, 0.5, 1, 5, 10, 30, 60, 300, 600, 1800, 3600, 14400 ]`.
+
+<!-- semconv metric.messaging.latency.duration(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `messaging.latency.duration` | Histogram | `s` | Measures the observed duration between when a message was created and when it began being processed. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+<!-- endsemconv -->
+
+### Metric: `metric.messaging.buffering.duration`
```

Comment:
I've read the descriptions to `metric.messaging.latency.duration` and `metric.messaging.buffering.duration` twice and I'm still not 100% sure I understand the difference between them (or why this one is called `buffering`)

I think `buffering` is the time between when the event was written to the kafka topic and the time that it was read by the consumer.

And `latency` is `buffering` plus the time that it took for the producer to write the message to the topic, right?

If this is correct, there are terms that confluent uses that I think I prefer. (from: https://www.confluent.io/blog/configure-kafka-to-minimize-latency/)

The term `end-to-end latency` instead of `buffering`:

> "End-to-end latency" is the time between when the application logic produces a record via KafkaProducer.send() to when the record can be consumed by the application logic via KafkaConsumer.poll().

They don't have as clear of a name for what we're calling `metric.messaging.latency.duration`, but they refer to the period I think we're trying to measure here as "catch up latency".  

That feels more descriptive to me than just `latency` as it's how far the consumer needs to go to "catch up" to current messages on the topic.

Here's the specific diagram from confluent that I'm referencing:

![CleanShot 2024-05-16 at 15 36 55](https://github.com/open-telemetry/semantic-conventions/assets/5231/96f5e1c5-57ff-4956-8a46-684420178a87)